File: src/main/java/com/example/app/service/impl/NoteServiceImpl.java
package com.example.app.service.impl;

import com.example.app.repository.NoteRepository;
import com.example.app.service.NoteService;
import com.example.app.utils.model.entities.Note;
import lombok.AllArgsConstructor;
import lombok.NonNull;
import org.springframework.stereotype.Service;

import java.util.List;


// TODO: 3/18/23  Сделать проверка существования заметки перед удалением или обновлением
@Service
@AllArgsConstructor
public class NoteServiceImpl implements NoteService {

    NoteRepository noteRepository;
    @Override
    public Note addNote(@NonNull Note note) {
        return noteRepository.save(note);

    }

    @Override
    public Note updateNote(@NonNull Note note) {
        return noteRepository.updateNote(note.getText(), note.getId());
    }

    @Override
    public void deleteNote(@NonNull Long id) {
        noteRepository.deleteById(id);
    }

    @Override
    public List<Note> getAllNotes() {
        String owner = "";
        return noteRepository.getNotesByOwner(owner);
    }
}

File: src/main/kotlin/com/example/service/impl/NoteServiceImpl.kt
package com.example.service.impl

import com.example.repository.NoteRepository
import com.example.service.NoteService
import com.example.utils.exceptions.notesExceptions.NoteCannotBeAddedException
import com.example.utils.exceptions.notesExceptions.NoteCannotUpdatedException
import com.example.utils.dto.entities.Note
import org.springframework.stereotype.Service

// TODO: 3/18/23  Сделать проверка существования заметки перед удалением или обновлением
@Service
class NoteServiceImpl(
    private val noteRepository: NoteRepository
) : NoteService {

    override fun addNote(note: Note): Note {
        return noteRepository.save(note) ?: throw NoteCannotBeAddedException()
    }

    override fun updateNote(note: Note): Note{
        return noteRepository.updateNote(note.getText(), note.getId()) ?: throw NoteCannotUpdatedException()
    }

    override fun deleteNote(id: Long) {
        noteRepository.deleteById(id)
    }

    override fun allNotes(): List<Note> {
        // TODO: добавить owner 
        val owner = ""
        return noteRepository.getNotesByOwner(owner)
    }
}
